service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/RemoveOrderControllerSubmittedEvent.java

```diff
@@ -141,5 +199,176 @@ void shouldNotPublishPopulateStandardDirectionsEventIfASDOHasNotBeenRemoved() {
         postSubmittedEvent(callbackRequest);
 
         verifyNoMoreInteractions(coreCaseDataService);
+        checkThat(() -> verify(notificationClient, never())
+            .sendEmail(eq(SDO_REMOVAL_NOTIFICATION_TEMPLATE), any(), any(), any()));
+    }
+
+    @Test
+    void shouldNotSendEmailNotificationsIfABlankOrderIsRemoved() {
+        Element<GeneratedOrder> hiddenBlankOrder = element(buildBlankOrder());
+
+        CaseDetails caseDetails = CaseDetails.builder()
+            .data(Map.of(
+                "hiddenOrders", singletonList(element(hiddenBlankOrder)),
```

Comment:
This doesn't look right, this will make an `List<Element<Element<GeneratedOrder>>`. I advise using CaseData class instead as suggested before so that you can ensure type safety

Reply:
it was a mistake. Fixed it now